### ISSUE TITLE
[no-explicit-any] Add `ignore-rest-args` equivalent option from tslint's `no-any`

### ISSUE THREAD

**Issue by Dilatorily** [typescript-eslint/typescript-eslint#397]
**Repro**

```JSON
{
  "rules": {
    "typescript/no-explicit-any": "error"
  }
}
```

```TS
function foo(a: number, ...rest: any[]): void {
    return;
}
```

**Expected Result**
`no-explicit-any` has a setting to ignore rest parameters.

**Actual Result**
`no-explicit-any` does not have any settings to configure rest parameters.

**Additional Info**
https://palantir.github.io/tslint/rules/no-any/

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.5.0` |
| `@typescript-eslint/parser`        | `1.5.0` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.3` |
| `node`                             | `11.6.0` |
| `npm`                              | `6.5.0` |

**Comment by jindal93** [comment]
Any updates on this?

**Comment by bradzacher** [comment]
Nope! I've been prioritising bugfixes over new features.

Happy to accept PRs - love more contributors!
